### PR TITLE
Fix/refactor tracker

### LINKED ISSUE #163: Refactor type and parameter names to match python
## Issue

**Description**
_A clear and concise description of the issue._

- [x] https://github.com/microsoft/electionguard-python/issues/327
- [x] https://github.com/microsoft/electionguard-python/issues/326
- [x] https://github.com/microsoft/electionguard-python/issues/324
- [x] https://github.com/microsoft/electionguard-python/issues/322
- [x] https://github.com/microsoft/electionguard-python/pull/348

### PR DESCRIPTION
**Pull request by AddressXception**
_🚨 Please review the [guidelines for contributing](../CONTRIBUTING.md) to this repository._

### Issue
*Link your PR to an issue*

https://github.com/microsoft/electionguard-python/issues/322
https://github.com/microsoft/electionguard-python/pull/348
https://github.com/microsoft/electionguard-python/pull/339
https://github.com/microsoft/electionguard-python/pull/338
https://github.com/microsoft/electionguard-python/pull/337

Fixes #163

### Description
see individual tickets and PR's from python for details

### Testing
Run the tests

### CODE REVIEW

File: src/electionguard/facades/ballot_code.cpp [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        *out_device_hash = AS_TYPE(eg_element_mod_q_t, hash.release());
+        return ELECTIONGUARD_STATUS_SUCCESS;
+    } catch (const exception &e) {
+        Log::error(":eg_ballot_code_get_hash_for_device", e);
+        return ELECTIONGUARD_STATUS_ERROR_RUNTIME_ERROR;
+    }
+}
+
+eg_electionguard_status_t
+eg_ballot_code_get_rotating_ballot_code(eg_element_mod_q_t *in_previous, uint64_t in_timestamp,
```
Comment (keithrfung) [inline src/electionguard/facades/ballot_code.cpp:36]: This should be changed and the `rotating` word should be removed.

File: src/electionguard/facades/encrypt.cpp [hunk 2]
```diff
@@ -242,17 +240,17 @@ eg_electionguard_status_t eg_encrypt_contest(eg_plaintext_ballot_contest_t *in_p
...
                                             eg_internal_election_description_t *in_metadata,
                                             eg_ciphertext_election_context_t *in_context,
-                                            eg_element_mod_q_t *in_seed_hash,
+                                            eg_element_mod_q_t *in_ballot_code_seed,
                                             bool in_should_verify_proofs,
                                             eg_ciphertext_ballot_t **out_handle)
 {
     try {
         auto *plaintext = AS_TYPE(PlaintextBallot, in_plaintext);
         auto *metadata = AS_TYPE(InternalElectionDescription, in_metadata);
```
Comment (keithrfung) [inline src/electionguard/facades/encrypt.cpp:249]: Another reference to Internal Election Description

File: src/electionguard/facades/encrypt.cpp [hunk 3]
```diff
@@ -292,17 +290,17 @@ eg_electionguard_status_t eg_encrypt_ballot_with_nonce(
...
                                                     eg_internal_election_description_t *in_metadata,
                                                     eg_ciphertext_election_context_t *in_context,
-                                                    eg_element_mod_q_t *in_seed_hash,
+                                                    eg_element_mod_q_t *in_ballot_code_seed,
                                                     bool in_should_verify_proofs,
                                                     eg_compact_ciphertext_ballot_t **out_handle)
 {
     try {
         auto *plaintext = AS_TYPE(PlaintextBallot, in_plaintext);
         auto *metadata = AS_TYPE(InternalElectionDescription, in_metadata);
```
Comment (keithrfung) [inline src/electionguard/facades/encrypt.cpp:299]: InternalElectionDescription again.

File: test/electionguard/test_words.cpp [hunk 4]
```diff
@@ -1,40 +1,94 @@
...
+    {
+        // Arrange
+        auto deviceHash =
+          BallotCode::getHashForDevice(12345UL, 23456UL, 34567UL, "some-location-string");
+        const auto &ballotHash = ONE_MOD_Q();
+        const auto &anotherBallotHash = TWO_MOD_Q();
+        uint64_t timestamp = 1000;
+        auto firstRotatingHash =
+          BallotCode::getRotatingBallotCode(*deviceHash, timestamp, ballotHash);
+        auto secondRotatingHash =
```
Comment (keithrfung) [inline test/electionguard/test_words.cpp:58]: The use of rotating hash here should likely get removed. Implies chaining.

**Review by keithrfung — CHANGES_REQUESTED** [review]
The most notable fix is `InternalElectionDescription` reference, but a couple other small things.

File: bindings/netstandard/ElectionGuard/ElectionGuard.Shared/Encrypt.cs [hunk 5]
```diff
@@ -118,53 +118,53 @@ public class Encrypt
...
+                    Console.WriteLine($"EncryptWithNonce Ballot Error Status: {status}");
                 }
                 return new CiphertextBallot(ciphertext);
             }
         }
 
         public static unsafe CompactCiphertextBallot CompactBallot(
             PlaintextBallot ballot,
-            InternalElectionDescription metadata,
+            InternalElectionDescription internalManifest,
```
[the remark this answers is not in the document]
Comment (AddressXception) [reply bindings/netstandard/ElectionGuard/ElectionGuard.Shared/Encrypt.cs:155]: next pr

File: bindings/netstandard/ElectionGuard/ElectionGuard.Shared/NativeInterface.cs [hunk 6]
```diff
@@ -1022,51 +1023,52 @@ protected override bool Free()
...
+            //     CiphertextElectionContext.CiphertextElectionContextHandle context,
+            //     ElementModQ.ElementModQHandle ballot_code_seed,
+            //     ElementModQ.ElementModQHandle nonce,
+            //     bool shouldVerifyProofs,
+            //     out CiphertextBallot.CiphertextBallotHandle handle);
+
+            // [DllImport(DllName, EntryPoint = "eg_encrypt_contest")]
+            // internal static extern Status Contest(
+            //     PlaintextBallotContest.PlaintextBallotContestHandle plaintext,
+            //     ContestDescription.InternalElectionDescriptionHandle metadata,
```
[the remark this answers is not in the document]
Comment (AddressXception) [reply bindings/netstandard/ElectionGuard/ElectionGuard.Shared/NativeInterface.cs:1040]: next pr

File: test/electionguard/mocks/words.cpp [hunk 7]
```diff
@@ -5419,4 +5421,61 @@ namespace electionguard
         return -1;
     }
 
-} // namespace electionguard
+    EG_API int16_t getIndex(const char *word) { return getIndex(string(word)); }
+
+    EG_API string hashToWords(const ElementModQ &trackerHash, const char *separator)
```
[the remark this answers is not in the document]
Comment (AddressXception) [reply test/electionguard/mocks/words.cpp]: i just deleted the files.  theyll be in history if we need them

[the remark this answers is not in the document]
Comment (rc-ms) [reply test/electionguard/mocks/words.cpp]: no worries we can reconstitute later if need be

**Review by keithrfung — APPROVED** [review]

**Pull request merged**